src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc

```diff
@@ -0,0 +1,1505 @@
+----
+
+Gradle::
++
+[source,groovy]
+----
+// We assume you use `log4j-bom` for dependency management
+runtimeOnly 'org.apache.logging.log4j:log4j-jdbc-dbcp2'
+----
+
+====
+
+xref:plugin-reference.adoc#org-apache-logging-log4j_log4j-jdbc-dbcp2_org-apache-logging-log4j-core-appender-db-jdbc-PoolingDriverConnectionSource[{open-book} Plugin reference for `PoolingDriver`]
+
+[#JdbcAppender-MapMessage]
+=== Map Message handling
+
+If the optional <<JdbcAppender-element-Layout,nested element of type `Layout<? Extends Message>`>> is provided, log events containing messages of type
+xref:manual/messages.adoc#MapMessage[`MapMessage`]
+will be treated specially.
+For each <<ColumnMapping,column mapping>> (except those containing literals) the <<attr-source,`source`>> attribute will be used as key to the value in `MapMessage` that will be stored in column <<attr-name,`name`>>.
+
+[#JdbcAppender-examples]
+=== Configuration examples
+
+Here are a couple of sample configurations for the JDBCAppender, as well as a sample factory implementation that uses Commons Pooling and Commons DBCP to pool database connections:
+
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<Configuration status="error">
+  <Appenders>
+    <JDBC name="databaseAppender" tableName="dbo.application_log">
+      <DataSource jndiName="java:/comp/env/jdbc/LoggingDataSource" />
+      <Column name="eventDate" isEventTimestamp="true" />
+      <Column name="level" pattern="%level" />
+      <Column name="logger" pattern="%logger" />
+      <Column name="message" pattern="%message" />
+      <Column name="exception" pattern="%ex{full}" />
+    </JDBC>
+  </Appenders>
+  <Loggers>
+    <Root level="warn">
+      <AppenderRef ref="databaseAppender"/>
+    </Root>
+  </Loggers>
+</Configuration>
+----
+
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<Configuration status="error">
+  <Appenders>
+    <JDBC name="databaseAppender" tableName="LOGGING.APPLICATION_LOG">
+      <ConnectionFactory class="net.example.db.ConnectionFactory" method="getDatabaseConnection" />
+      <Column name="EVENT_ID" literal="LOGGING.APPLICATION_LOG_SEQUENCE.NEXTVAL" />
+      <Column name="EVENT_DATE" isEventTimestamp="true" />
+      <Column name="LEVEL" pattern="%level" />
+      <Column name="LOGGER" pattern="%logger" />
+      <Column name="MESSAGE" pattern="%message" />
+      <Column name="THROWABLE" pattern="%ex{full}" />
+    </JDBC>
+  </Appenders>
+  <Loggers>
+    <Root level="warn">
+      <AppenderRef ref="databaseAppender"/>
+    </Root>
+  </Loggers>
+</Configuration>
+----
+
+The following configuration uses no layout to indicate that the appender should match the keys of a `MapMessage` to the names of
+``ColumnMapping``s when setting the values of the Appender's SQL INSERT statement.
+This lets you insert rows for custom values in a database table based on a Log4j `MapMessage` instead of values from `LogEvent`.
+
+[source,xml]
+----
+<Configuration status="debug">
```

Comment:
This needs to be ported to `examples` in all supported configuration formats.

Reply:
Fixed in https://github.com/apache/logging-log4j2/pull/2790/commits/fad8d7f8603086a8abaa44c0c9e2d1dd6d7524fe